src/io/json/read/deserialize.rs

```diff
@@ -43,13 +59,15 @@ fn build_extract(data_type: &DataType) -> Extract {
             Box::new(move |value| {
                 let integer = match value {
                     Value::Number(number) => Some(deserialize_int_single::<i64>(*number)),
-                    Value::Bool(number) => Some(i64::from(*number)),
+                    Value::Bool(number) => Some(Ok(i64::from(*number))),
```

Comment:
This changes type inside `Option`, so it either didn't work correctly earlier, or won't work now.
Or there is something else that knows how to coerce `Result<T>` into `T`?